src/audio_excl.rs

```diff
@@ -16,70 +15,66 @@ pub struct FileLister {
 }
 
 impl FileLister {
-    pub fn new<T: AsRef<Path>>(path: T, timeout: &Duration) -> io::Result<FileLister> {
-        let mut rng = rand::thread_rng();
-        let num_items = fs::read_dir(&path)?.count();
-        let items = fs::read_dir(&path)?
-            .filter_map(io::Result::ok)
+    pub fn new<P: AsRef<Path>>(path: P, timeout: Duration) -> io::Result<FileLister> {
+        let mut files = fs::read_dir(&path)?
+            .filter_map(Result::ok)
             .map(|s| s.path())
-            .choose_multiple(&mut rng, num_items);
-        println!("Got items in order {:?}", items);
-        return Ok(FileLister {
-            files: items,
-            lock_acq: HashMap::new(),
-            security: HashMap::new(),
-            timeout: *timeout,
-        });
+            .collect::<Vec<_>>();
+        files.shuffle(&mut rand::thread_rng());
+        println!("Got items in order {:?}", files);
+        Ok(FileLister {
+            files,
+            timeout,
+            ..Default::default()
+        })
     }
 
-    pub fn clean(&mut self) -> io::Result<()> {
```

Comment:
This function never actually returned an error, so I just made it return `()` instead.